src/main/java/Duke.java

```diff
@@ -1,10 +1,111 @@
+import task.*;
+
+import java.io.*;
+import java.text.ParseException;
+
 public class Duke {
+
+    private static final String FILEPATH = "duke.txt";
+    private Ui ui;
+    private Storage storage;
+    private TaskList taskList;
+
+    /**
+     * Constructor for Duke class.
+     * @param filePath Path for the Duke save file.
+     */
+    public Duke(String filePath) {
+        //storage = new Storage(filePath);
+        taskList = Storage.loadTasks(FILEPATH);
+        ui = new Ui();
+    }
+
+    /**
+     * Method to run the Duke program.
```

Comment:
Perhaps you could try to short this run method? It seems to be longer than 30 LOC which could make it difficult to debug. Just a suggestion.